lib/ansible/template/__init__.py

```diff
@@ -726,6 +726,9 @@ def do_template(self, data, preserve_trailing_newlines=True, escape_backslashes=
                 else:
                     return data
 
+            # jinja2 global is inconsistent across versions, this normalizes them
+            t.globals['dict'] = dict
```

Comment:
You may want to consider making a wrapper function that checks the jinja2 version before overriding.  The reason being that jinja2 could enhance their dict function in the future at which point we would be providing less features rather than more.